pkg/controller/kieapp/shared/utils.go

```diff
@@ -179,14 +179,11 @@ func envVarEqual(env corev1.EnvVar, envList []corev1.EnvVar) bool {
 
 func EnvOverride(dst, src []corev1.EnvVar) []corev1.EnvVar {
 	for _, cre := range src {
-		for envIndex, e := range dst {
-			if isEnvVarInList(cre.Name, dst) {
-				if cre.Name == e.Name {
-					dst[envIndex] = cre
-				}
-			} else {
-				dst = append(dst, cre)
-			}
+		pos, _ := getEnvVar(cre.Name, dst)
```

Comment:
does this change how the function works? was this a fix of some kind?

Reply:
It’s just avoiding an extra iteration. The outcome is the same